lib/beaker/dsl/ezbake_utils.rb

```diff
@@ -145,74 +207,202 @@ def install_from_ezbake host, project_name, project_param_string, env_args={}, e
+              end
+              raise RuntimeError, error_message
+            end
+          end
         end
+      end
 
-        # untar tarball on host
-        on host, "tar -xzf " + remote_tarball
+      # Return the ezbake config.
+      #
+      # @return [Hash] configuration for ezbake, usually from ezbake.rb
+      # @api private
+      def ezbake_config
+        EZBakeUtils.config
+      end
+
+      # Prepares a staging directory for the specified project.
+      #
+      # @param [String] project_name The name of the ezbake project being worked
+      #   on.
+      # @param [String] project_param_string Parameters to be passed to ezbake
+      #   on the command line.
+      # @param [String] ezbake_dir The local directory where the ezbake project
+      #   resides or should reside if it doesn't exist already.
+      # @api private
+      def ezbake_stage project_name, project_param_string=nil,
+                       ezbake_dir="tmp/ezbake"
+        ezbake_tools_available?
+        #conditionally_clone "gitmirror@github.delivery.puppetlabs.net:puppetlabs-ezbake.git",
+        #                    ezbake_dir
+        # TODO: will need removal before merging
+        conditionally_clone "git@github.com:kbarber/ezbake.git",
+                            ezbake_dir, "pdb-1034-ezbake-pr-testing"
+
+        # If it was not already passed through, try retreiving the project
+        # version from the local repository in cwd.
+        if project_param_string.nil?
+          project_param_string = "#{project_name}-version=#{ezbake_lein_project_version}"
+        end
+
+        # Get the absolute path to the local repo
+        m2_repo = File.join(Dir.pwd, 'tmp', 'm2-local')
+
+        lein_prefix = 'lein update-in : assoc :local-repo "\"' + m2_repo + '\"" --'
+        ezbake_local_cmd "#{lein_prefix} install", :throw_on_failure => true
+
+        Dir.chdir(ezbake_dir) do
+          ezbake_local_cmd "#{lein_prefix} run -- stage #{project_name} #{project_param_string}",
+                           :throw_on_failure => true
+        end
+
+        staging_dir = File.join(ezbake_dir, 'target', 'staging')
+        Dir.chdir(staging_dir) do
+          ezbake_local_cmd 'rake package:bootstrap'
+
+          load 'ezbake.rb'
+          ezbake = EZBake::Config
+          ezbake[:package_version] = `echo -n $(rake pl:print_build_param[ref] | tail -n 1)`
+          EZBakeUtils.config = ezbake
+        end
+      end
+
+      # Executes a local command using system, logging the prepared command
+      #
+      # @param [String] cmd command to execute
+      # @param [Hash] opts options
+      # @option opts [bool] :throw_on_failure If true, throws an
+      #   exception if the exit code is non-zero. Defaults to false.
+      # @return [bool] true if exit == 0 false if otherwise
+      # @raise [RuntimeError] if :throw_on_failure is true and
+      #   command fails
+      # @api private
+      def ezbake_local_cmd cmd, opts={}
```

Comment:
I would agree that this is a bigger issue (a simple wrapper around system is never simple :) ).  This might be a better option as a helper library instead of part of mainline beaker - until we decide what we are trying to accomplish and do a proper build out.